Remove Cairo dependency for BackingStore

donny-dont opened the pull request:
#### 6619d9b70306f0cdf01f83b058ab995df16a4480
<pre>
Remove Cairo dependency for BackingStore
<a href="https://bugs.webkit.org/show_bug.cgi?id=269495">https://bugs.webkit.org/show_bug.cgi?id=269495</a>

Reviewed by Adrian Perez de Castro.

Both `USE(GRAPHICS_LAYER)` and `USE(CAIRO)` for non-WPE platforms use instances
of `BackingStore`. To prepare for a `USE(SKIA)` implementation of it make the
code more cross-platform by using platform agnostic definitions. Rename and
move files accordingly to fit conventions.

Actual Skia implementation will come later.

* Source/WebKit/Platform/WC.cmake:
* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformPlayStation.cmake:
* Source/WebKit/PlatformWin.cmake:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/UIProcess/BackingStore.h: Renamed from Source/WebKit/UIProcess/cairo/BackingStore.h.
* Source/WebKit/UIProcess/CoordinatedGraphics/DrawingAreaProxyCoordinatedGraphics.cpp:
* Source/WebKit/UIProcess/CoordinatedGraphics/DrawingAreaProxyCoordinatedGraphics.h:
* Source/WebKit/UIProcess/cairo/BackingStoreCairo.cpp: Renamed from Source/WebKit/UIProcess/cairo/BackingStore.cpp.
* Source/WebKit/UIProcess/wc/DrawingAreaProxyWC.cpp:
* Source/WebKit/UIProcess/wc/DrawingAreaProxyWC.h:

Canonical link: <a href="https://commits.webkit.org/274823@main">https://commits.webkit.org/274823@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0dfe2d981cbd7a04b1c6bf2b64d4dfdf0e9ef75f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/40044 "3 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/19056 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/42420 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/42589 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/36011 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/21965 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/16385 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/33326 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/40618 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/16044 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/34590 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/13877 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/13908 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/43867 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/36332 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/35879 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/39642 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/14859 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/12196 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/37910 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/16478 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9002 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/16527 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/16122 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/WebKit/UIProcess/gtk/AcceleratedBackingStore.h:
```diff
@@ -54,7 +54,7 @@ class AcceleratedBackingStore {
     virtual void snapshot(GtkSnapshot*) = 0;
 #else
-    virtual bool paint(cairo_t*, const WebCore::IntRect&) = 0;
+    virtual bool paint(PlatformPaintContextPtr, const WebCore::IntRect&) = 0;
```
I think you forgot to include `BackingStore.h` here, so the GTK builders are failing because this translation unit is not seeing the `PlatformPaintContextPtr` type definition. Maybe the easiest is to add something like the following above:

```cpp
#if USE(CAIRO)
typedef struct _cairo cairo_t;
using PlatformPaintContextPtr = cairo_t*;
#endif
```

I think this should be enough to fix the GTK build issue.

donny-dont replied:
Doh! I thought `AcceleratedBackingStore` descended from `BackingStore` which is not the case. Think I'm just going to roll back the changes in that class.

aperezdc reviewed (changes requested):
GTK build failing, there's a missing type definition (at least).

@donny-dont Could you take a look at it?

aperezdc reviewed (approved):
Patch LGTM now, but please fix the style checker complaints before landing. Thanks!

donny-dont committed:
Remove Cairo dependency for BackingStore
https://bugs.webkit.org/show_bug.cgi?id=269495

Reviewed by Adrian Perez de Castro.

Both `USE(GRAPHICS_LAYER)` and `USE(CAIRO)` for non-WPE platforms use instances
of `BackingStore`. To prepare for a `USE(SKIA)` implementation of it make the
code more cross-platform by using platform agnostic definitions. Rename and
move files accordingly to fit conventions.

Actual Skia implementation will come later.

* Source/WebKit/Platform/WC.cmake:
* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformPlayStation.cmake:
* Source/WebKit/PlatformWin.cmake:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/UIProcess/BackingStore.h: Renamed from Source/WebKit/UIProcess/cairo/BackingStore.h.
* Source/WebKit/UIProcess/CoordinatedGraphics/DrawingAreaProxyCoordinatedGraphics.cpp:
* Source/WebKit/UIProcess/CoordinatedGraphics/DrawingAreaProxyCoordinatedGraphics.h:
* Source/WebKit/UIProcess/cairo/BackingStoreCairo.cpp: Renamed from Source/WebKit/UIProcess/cairo/BackingStore.cpp.
* Source/WebKit/UIProcess/wc/DrawingAreaProxyWC.cpp:
* Source/WebKit/UIProcess/wc/DrawingAreaProxyWC.h:

Canonical link: https://commits.webkit.org/274823@main
